src/Plugins/RpcServer/RpcServer.Wallet.cs

```diff
@@ -555,6 +578,27 @@ private JObject SignAndRelay(DataCache snapshot, Transaction tx)
             }
         }
 
+        /// <summary>
+        /// Signs transaction without relaying.
+        /// </summary>
+        /// <param name="snapshot">The data snapshot.</param>
+        /// <param name="tx">The transaction to sign and relay.</param>
+        /// <returns>A JSON object containing the transaction details.</returns>
+        private JObject SignAndNoRelay(DataCache snapshot, Transaction tx)
+        {
+            ContractParametersContext context = new(snapshot, tx, settings.Network);
+            wallet.Sign(context);
+            if (context.Completed)
+            {
+                tx.Witnesses = context.GetWitnesses();
+                return Utility.TransactionToJson(tx, system.Settings);
```

Comment:
Have you considered reusing `ContractParametersContext` for this? It's more flexible to me since it allows to store/exchange partially signed transactions as well.